test/functional/lua/diagnostic_spec.lua

```diff
@@ -869,4 +869,80 @@ describe('vim.diagnostic', function()
       assert(loc_list[1].lnum < loc_list[2].lnum)
     end)
   end)
+
+  describe('match()', function()
+    it('matches a string', function()
```

Comment:
Could you also add a test for a case where the pattern doesn't match a string? (Returning an empty list?)